File: EmployeeKnockOut/wwwroot/js/App/Employee.Model.js
/// <reference path="../knockoutjs3.5.1.js" />
 

 
function EmployeeModel(item) {
    let self = this;
    item = item || {};
    self.id = ko.observable(item.id || 0);
    self.name = ko.observable(item.name || '');
    self.gender = ko.observable(item.gender || 'Male')
    self.address = ko.observable(item.address || '');
    self.department = ko.observable(item.department || '');
    self.salary = ko.observable(item.salary || '');


}

File: EmployeeKnockOut/Models/EmployeeModel.cs
using System.ComponentModel.DataAnnotations;

namespace EmployeeKnockOut.Models
{
    public class EmployeeModel
    {public int Id { get; set; }
        [Required]
        public string Name { get; set; }   

        public string Gender { get; set; }
        [Required]
        public string Address { get; set; }
        [Required]
        public string Department { get; set; }

        [RegularExpression("([1-9][0-9]*)",ErrorMessage ="Please Enter Valid amount of Salary")]
        [Required]
        public string Salary { get; set; }  

    }
}
